packages/app/components/settings/settings-wallet-item.tsx

```diff
@@ -0,0 +1,209 @@
+    }
+    if (wallet.phone_number && wallet.is_phone) {
+      return (
+        <LoginWithPlatformTitle
+          text={wallet.phone_number}
+          Icon={PhonePortraitOutline}
+        />
+      );
+    }
+    if (wallet.is_google) {
+      return (
+        <LoginWithPlatformTitle
+          text={"Login from Google"}
+          Icon={GoogleOriginal}
+        />
+      );
+    }
+    if (wallet.is_apple) {
+      return <LoginWithPlatformTitle text={"Login from Apple"} Icon={Apple} />;
+    }
+    return null;
+  }, [
+    wallet.email,
+    wallet.is_apple,
+    wallet.is_email,
+    wallet.is_google,
+    wallet.is_phone,
+    wallet.phone_number,
+  ]);
+
+  const isHasSubTitle =
+    wallet.email || wallet.phone_number || wallet.is_google || wallet.is_apple;
+
+  if (!nickname && !isHasSubTitle) {
+    return null;
+  }
+  return (
+    <View tw="mb-2.5 flex-row flex-wrap items-center">
+      {Boolean(nickname) && (
+        <Text tw="mr-2 text-base font-bold text-gray-900 dark:text-white">
+          {nickname}
+        </Text>
+      )}
+      {Boolean(nickname) && isHasSubTitle ? (
+        <View tw="flex-row items-center justify-center">
+          <Text tw="text-sm text-gray-900 dark:text-white">(</Text>
+          {Title}
+          <Text tw="text-sm text-gray-900 dark:text-white">)</Text>
+        </View>
+      ) : (
+        Title
+      )}
+    </View>
+  );
+};
+
+export const SettingsWalletItem = (props: Props) => {
+  const address = props.wallet.address;
+  const ensDomain = props.wallet.ens_domain;
+  const nickname = props.wallet.nickname;
+  const wallet = props.wallet;
+  const { userAddress } = useCurrentUserAddress();
+  const user = useUser();
+  const isDark = useIsDarkMode();
+  const display = ensDomain ? ensDomain : formatAddressShort(address);
+  const isEthereumAddress = address.startsWith("0x");
+
+  const isConnectedAddress =
+    userAddress?.toLowerCase() === address?.toLowerCase();
+
+  const isPrimary = user.user?.data.profile.primary_wallet?.address === address;
+  const { setPrimaryWallet } = useSetPrimaryWallet();
+  const MakePrimaryBtn = useMemo(() => {
+    return (
+      <View tw="mb-2 md:mb-0">
+        {!isPrimary ? (
+          <PressableHover
+            tw="mr-4 h-6 flex-row items-center justify-center self-start rounded-3xl border bg-black px-2 dark:bg-white md:h-8"
+            onPress={() => setPrimaryWallet(props.wallet)}
+          >
+            <Text tw="text-xs text-white dark:text-gray-900">Make Primary</Text>
+          </PressableHover>
+        ) : (
+          <PressableHover tw="mr-4 h-6 flex-row items-center justify-center self-start rounded-3xl border border-green-500 bg-green-500/20 px-2 md:h-8">
+            <Check color={colors.green[700]} width={16} height={16} />
+            <Text tw="text-xs text-green-600"> Primary</Text>
+          </PressableHover>
+        )}
+      </View>
+    );
+  }, [isPrimary, props.wallet, setPrimaryWallet]);
```

Comment:
I don't know if you guys agreed to this type of component inside of component declaration, but it's actually anti-pattern.

Reply:
that's true! I just moved these types of components outside.